src/AbstractDbEntityFetcher.php

```diff
@@ -91,6 +91,23 @@ protected function getFetchPaginationResult(array $objects, $pagination)
     }
 
     /**
+     * Helper method to get pagination result as objects if pagination is requested.
+     *
+     * @param array $rows
+     * @param bool  $pagination
+     * @return array
+     */
+    protected function getFetchPaginationAsDbEntitiesResult(array $rows, $pagination)
```

Comment:
As @jandreasn suggested, `getDbEntitiesFromRowsPaginated()` might be a better name for this method. Also if you send in `$pagination` as false, this method now does the exact same thing as `getDbEntitiesFromRows()`, so maybe we should remove that parameter all together?